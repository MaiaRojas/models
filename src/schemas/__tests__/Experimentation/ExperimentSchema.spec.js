const mongoose = require('mongoose/browser');
const { ExperimentSchema } = require('../../')(mongoose);

describe('ExperimentSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, ExperimentSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate document with valid data whith minimal data', () => {
    const doc = new mongoose.Document({
      uid: '45orQ7aPbbPvF2fvMgbA1TsTFPo2',
      createdBy: mongoose.Types.ObjectId('5ca30d9cdd9c6f29c7d7b591'),
      slug: 'text',
      name: 'text',
      description: 'prueba text',
    }, ExperimentSchema);
    return doc.validate();
  });

  it('should validate document with valid data whith full data', () => {
    const doc = new mongoose.Document({
      uid: '45orQ7aPbbPvF2fvMgbA1TsTFPo2',
      createdBy: mongoose.Types.ObjectId('5ca30d9cdd9c6f29c7d7b591'),
      slug: 'text',
      name: 'text',
      description: 'prueba text',
      active: false,
      lastVersion: 2,
    }, ExperimentSchema);
    return doc.validate();
  });
});
