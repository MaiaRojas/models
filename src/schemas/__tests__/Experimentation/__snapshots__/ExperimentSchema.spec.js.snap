// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentSchema should fail validation when fields missing 1`] = `
Object {
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "description": [ValidatorError: Path \`description\` is required.],
  "name": [ValidatorError: Path \`name\` is required.],
  "slug": [ValidatorError: Path \`slug\` is required.],
  "uid": [ValidatorError: Path \`uid\` is required.],
}
`;
